### ISSUE TITLE
[HOLD #296548] Transfer balance is not displaying error messages

### ISSUE THREAD

**Issue by marcochavezf** [Expensify/App#12911]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
This issue surfaced after testing the new testing debit cards from this [PR](https://github.com/Expensify/Auth/pull/7245). We should also add the new `shouldForceUsingSandbox` flag to the functions `createPullTransaction` and `createPushTransaction` [here](https://github.com/Expensify/Auth/blob/2d2299028098d6ced66ba9d8a70e54106400e10b/auth/lib/Bancorp.cpp).

## Action Performed:
Break down in numbered steps
1. Go to Settings -> Payments -> Transfer balance
2. Select Instant debit card (add one of the valid [test debit cards](https://rftdevportal.thebancorp.com/reference/test-sandbox-cards))
3. Click on `Transfer $...`

## Expected Result:
Describe what you think should've happened
The error `Cannot transfer unsettled wallet funds to a debit card` should be displayed

## Actual Result:
Describe what actually happened
No error, the error is only shown in the console

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?
N/A

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 
**Reproducible in staging?:** 
**Reproducible in production?:** 
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:** 
**Issue reported by:** 
**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

https://user-images.githubusercontent.com/6829422/203191459-7eb9ddc1-4e1d-4bff-81b5-1232c3f79522.mov

**Comment by zanyrenney** [comment]
Hey @marcochavezf I'll test this tomorrow and see if i can reproduce 👍🏼

**Comment by zanyrenney** [comment]
Hey @marcochavezf I can't reproduce this because:
- I have no existing balance in my wallet (my wallet is also not active in my test account)
- when I attempt to send money from one test account to another, the only currency I can select is GBP
- I am prompted to settle up elsewhere (meaning the wallet balance doesn't increase) 

A few screenshots to explain what I mean:
<img width="1637" alt="2022-11-23_15-30-43" src="https://user-images.githubusercontent.com/56830058/203587877-43b1d5fa-e796-43d5-a620-f261ca42bd33.png">

<img width="1643" alt="2022-11-23_15-30-56" src="https://user-images.githubusercontent.com/56830058/203587888-819c1a65-d8cd-4e65-b3ae-14932df76fe9.png">

<img width="1648" alt="2022-11-23_15-30-17" src="https://user-images.githubusercontent.com/56830058/203587853-7b24d5fc-edc1-4d18-a4ef-266e7bb863a3.png">

I'll add `needs reproduction` label and bump the slack thread where Jason mentioned this issue to get more 👀 

Thanks!

**Comment by marcochavezf** [comment]
> The sandbox test cards don't have the full information I need to add a payment method

Ah I forgot to mention that we can use any expiration date and CVV while using any of the Bancorp sandbox debit cards. Also, I realized this is only reproducible locally because there's a [problem with the sandbox credentials](https://github.com/Expensify/Auth/pull/7245#issuecomment-1326813489) on staging, my apologies.

Since this bug will appear once we're able to add debit cards, I'm going to add the HOLD until we fix the sandbox credentials issue on staging.

**Comment by zanyrenney** [comment]
If this is on HOLD, shouldn't we make it weekly @marcochavezf ?

**Comment by zanyrenney** [comment]
Have updated the priority @marcochavezf - let me know if you disagree!

**Comment by zanyrenney** [comment]
Still on hold, right @marcochavezf ?

**Comment by marcochavezf** [comment]
Yup, we're waiting for Bancorp's response (more info [here](https://github.com/Expensify/Salt/pull/11676#issuecomment-1344585453)).

**Comment by marcochavezf** [comment]
No update here, we're still waiting for Bancorp's response.

**Comment by marcochavezf** [comment]
Still waiting on Bancorp's response. I'm going to use this [issue](https://github.com/Expensify/Expensify/issues/237181) to post updates about it.

**Comment by marcochavezf** [comment]
Latest update [here](https://github.com/Expensify/Expensify/issues/237181#issuecomment-1382250043)

**Comment by zanyrenney** [comment]
Thanks for the update @marcochavezf - from the linked issue, it looks like we are still waiting to hear[ back from Bancorp](https://github.com/Expensify/Expensify/issues/237181#issuecomment-1398590217)

**Comment by zanyrenney** [comment]
Any further update from Bancorp @marcochavezf ?

**Comment by marcochavezf** [comment]
Yup, there's a PR with the latest token in review. Hopefully, that will unblock this issue soon.

**Comment by marcochavezf** [comment]
No update, focused on EYOC MVP this week

**Comment by marcochavezf** [comment]
> No update, focused on EYOC MVP this week

**Comment by zanyrenney** [comment]
Thanks Marco! same again this week ? @marcochavezf

**Comment by marcochavezf** [comment]
Yup, last week and ECX week have been super busy

**Comment by zanyrenney** [comment]
Assume this is still on hold @marcochavezf ?

**Comment by marcochavezf** [comment]
Yes, it's still on hold. Working today on the integration tests for the bancorp sandbox PR

**Comment by zanyrenney** [comment]
Thanks @marcochavezf - keep the issue up to date as you go please. 🎉

**Comment by marcochavezf** [comment]
The testing debit cards on staging are working now. I will take a look at this one in spare cycles

**Comment by zanyrenney** [comment]
Hey @marcochavezf how are we doing with this one? If it's not a priority, let's update the label to monthly?

**Comment by marcochavezf** [comment]
Yeah that makes sense. Given the priority of some deadlines, I think this can be considered more as a Monthly. Changing it

**Comment by marcochavezf** [comment]
Hey, sorry for the delay in this one. I've been focused on the unauthenticated users in public rooms for RenderATL

**Comment by zanyrenney** [comment]
Hey @marcochavezf will you have more time to get to this now ExpensiCon is live?

**Comment by zanyrenney** [comment]
DM's @marcochavezf as this issue seems to be going overdue a lot.

**Comment by marcochavezf** [comment]
Ah sorry @zanyrenney! I'm focused on unauthenticated users in public rooms with a deadline for EOM. I'm planning to wrap up the implementation this week. Hopefully, I can continue with this one by EOW 🙇🏽‍♂️

**Comment by zanyrenney** [comment]
Going OOO and this is pretty active so needs management by bug0 team member, as per the [new process ](https://expensify.slack.com/archives/CK49UADB9/p1684182291511799)to stay assigned but assign another team member,  please can you keep a watch of this. I am OOO until 8th June. Thanks!

**Comment by isabelastisser** [comment]
For transparency, I have 28 daily issues on my pile and I might not be able to get to this today, but I will do my best to keep this going while you're out.

**Comment by isabelastisser** [comment]
Internal, no updates!

**Comment by isabelastisser** [comment]
Internal, no updates!

**Comment by isabelastisser** [comment]
On hold!

**Comment by isabelastisser** [comment]
I will be OOO until July 5, so assigning another team member. [SO](https://stackoverflowteams.com/c/expensify/questions/883) for reference.

I will keep @zanyrenney assigned because she was the original assignee.

**Comment by greg-schroeder** [comment]
Looks like Zany is OOO through 6/30, so I'll keep an eye on this until she returns :)

**Comment by greg-schroeder** [comment]
I'm changing this to `Weekly` if it's on hold

**Comment by marcochavezf** [comment]
No update, on hold

**Comment by greg-schroeder** [comment]
FYI Zany I'm going un-assign myself as you're back! Also this one is just sitting on hold anyway. :)

**Comment by zanyrenney** [comment]
Any further update on this one @marcochavezf ?

**Comment by marcochavezf** [comment]
The Bancorp sandbox token expired again. So this is on hold for the [new token](https://github.com/Expensify/Expensify/issues/296548#issuecomment-1634654580).

**Comment by marcochavezf** [comment]
There's a new sandbox token, but for some reason it's now working for me locally https://github.com/Expensify/Auth/pull/8332, I will try again

**Comment by marcochavezf** [comment]
Latest update about sandbox token [here](https://github.com/Expensify/Expensify/issues/296548#issuecomment-1658749134).

**Comment by zanyrenney** [comment]
Thanks @marcochavezf !

**Comment by zanyrenney** [comment]
Hey @marcochavezf what is the latest here please?

**Comment by marcochavezf** [comment]
We need a valid Bancorp token https://github.com/Expensify/Expensify/issues/296548, but we're focused on waves at the moment cc @cristipaval

**Comment by zanyrenney** [comment]
So... still on hold based on the above comment from @marcochavezf

**Comment by marcochavezf** [comment]
Ah no update here from my side, we're waiting on the new Bancorp token. @cristipaval since you were in contact with Bancorp, could you provide an update about the new token?

**Comment by zanyrenney** [comment]
@cristipaval please can we have an update?

**Comment by zanyrenney** [comment]
@cristipaval bump please?

**Comment by cristipaval** [comment]
I'll work with @deetergp this week to make Bancorp Sandbox work again. It hasn't been working for the last few months.

**Comment by marcochavezf** [comment]
Hi @cristipaval, I'm happy also to give a hand here since I updated it last time, did Bancorp already send the new token?

**Comment by cristipaval** [comment]
> Hi @cristipaval, I'm happy also to give a hand here since I updated it last time, did Bancorp already send the new token?

Thanks a lot, @marcochavezf, let's talk on the [issue](https://github.com/Expensify/Expensify/issues/296548) related to the sandbox token rotation.

**Comment by zanyrenney** [comment]
Hey @cristipaval / @marcochavezf do we have an update here please?

**Comment by cristipaval** [comment]
@deetergp emailed Bancorp for a new token, last week. Not sure if he has received one yet.

**Comment by zanyrenney** [comment]
should we assign @deetergp to this issue if work he is doing is impacting it?

**Comment by deetergp** [comment]
I still have not received the new token. I reached back out to Sean and his team to see where we are at.

**Comment by zanyrenney** [comment]
Thanks @deetergp can we bump them again please? Also should you be assigned to this issue @deetergp ?

**Comment by cristipaval** [comment]
@robertjchen is discussing with Bancorp team to whitelist our IPs. We need this changes now that we work with Cloudfare running in the backgorund. They opened an issue in their internal process and informed us that it usually takes 5-7 days. This happened yesterday.

**Comment by zanyrenney** [comment]
Thanks for the context @cristipaval @robertjchen any further insights here? Should we be making this monthly, there doesn't seem to have been a meaningful update since Aug 31, if its not being worked on weekly? It keeps going overdue 😅

**Comment by robertjchen** [comment]
We're still waiting on Bancorp- they're still figuring things on their end as of 9 hours ago, but we're close! If they don't figure it out this week we can demote

**Comment by zanyrenney** [comment]
oh no, why is this still showing as overdue.

**Comment by zanyrenney** [comment]
what's the update here @robertjchen @marcochavezf ?

**Comment by robertjchen** [comment]
Latest is happening here: https://github.com/Expensify/Expensify/issues/296548#issuecomment-1810030541

**Comment by zanyrenney** [comment]
Is this still on hold then? @robertjchen @marcochavezf

**Comment by marcochavezf** [comment]
Yeah, we're still waiting for the bancorp token to be updated

**Comment by deetergp** [comment]
The Bancorp and Tabapay people are still conversing with one another trying to figure out how to get us a new token. It's a little ridiculous.

**Comment by marcochavezf** [comment]
No update, focused on wave bugs

**Comment by marcochavezf** [comment]
No update

**Comment by zanyrenney** [comment]
what's the latest on this one please @marcochavezf ?

**Comment by marcochavezf** [comment]
Given our priorities in wave work, I will close it out. Feel free to re-open if you disagree.
